source/transactions/tests/README.rst

```diff
@@ -163,11 +200,18 @@ Then for each element in ``tests``:
    into the test collection, using writeConcern "majority".
 #. If ``failPoint`` is specified, its value is a configureFailPoint command.
    Run the command on the admin database to enable the fail point.
+
+   - When testing against a sharded cluster run this command on ALL mongoses.
```

Comment:
I assume this requires that drivers either:

 * provide some API to return all nodes within the topology (i.e. all known mongos nodes in the case of a shard cluster) and allow commands to explicitly use a selected mongos
 * have the test suite parse the connection string and extract each mongos host so that individual connections can be made for the purpose of configuring fail points

Reply:
Yes I think the second one makes the most sense. AFAIK there's no way to discover new mongoses from a cluster expect through the seed addresses in a connection string.